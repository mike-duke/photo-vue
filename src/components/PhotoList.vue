<template>
  <section class="photo-list">
    <h2>Viewing
      <span class="search-term">{{ searchTerm }}</span>
    </h2>
    <div id="photos">
      <Photo v-for="photo in photos" :photo="photo" :key="photo.id"></Photo>
    </div>
  </section>
</template>

<script>
  import Photo from './Photo.vue';

  export default {
    name: 'PhotoList',
    props: {
      photos: {
        type: Array,
        required: true
      },
      searchTerm: {
        type: String,
        required: true
      }
    },
    components: {
      Photo
    }
  }
</script>

<style lang="scss">
  .photo-list {
    display: flex;
    flex-direction: column;
    text-align: center;
    width: 100vw;
    background-attachment: fixed;
    background: rgba(0, 0, 0, 0.5);
    padding: 20px 0;
    font-size: 1.4rem;
    
    h2 {
      color: whitesmoke;
    }
  }

  #photos {
    display: flex;
    flex-direction: column;
    align-items: center;

    @media screen and (min-width: 400px) {
      flex-wrap: wrap;
      flex-direction: initial;
    } 
  }

  .search-term {
    color: lightgreen;
  }
</style>
