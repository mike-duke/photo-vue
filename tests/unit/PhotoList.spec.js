import PhotoList from '@/components/PhotoList';
import { mount } from '@vue/test-utils';

describe('PhotoList', () => {
  it('should match snapshot', () => {
    const wrapper = mount(PhotoList, {
      propsData: {
        photos: [],
        searchTerm: ''
      }
    });

    expect(wrapper).toMatchSnapshot();
  });
});