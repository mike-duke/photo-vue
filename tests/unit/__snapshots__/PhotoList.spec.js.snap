// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoList should match snapshot 1`] = `
<section class="photo-list">
  <h2>Viewing
    <span class="search-term"></span></h2>
  <div id="photos"></div>
</section>
`;
